pandas/core/groupby/groupby.py

```diff
@@ -758,6 +759,18 @@ def _try_cast(self, result, obj, numeric_only=False):
         else:
             dtype = obj.dtype
 
+        if is_extension_array_dtype(dtype):
```

Comment:
so i think you have to move this down. if the function is returning a scalar we don't care (e.g. around line 775 in this diff). Also you prob want to check if its an extension type, then try to convert it. I am not sure if we then need to convert it even more. I think some test cases will tell us.